### DISCUSSION TITLE
Where is the documentation stored?

### DISCUSSION THREAD

**Discussion by iach526526** [badges/shields#10474] [Q&A]
I'd like to help translate the Shield.io website, but I can't find the folder for the 'Badges' page. Where is it? Is it in the `docs` folder or in master/frontend?

**Comment by chris48s** [comment]
Translating the site is not really something we are looking for help with. We don't have the expertise to review or maintain translations. I posted a bit more detail about this in https://github.com/badges/shields/issues/10393#issuecomment-2242562253

---

That said, to answer your question the documentation isn't a stand-alone resource. It is quite tightly integrated with the code. Each service defines an Open API spec for itself and then the documentation is generated from those OpenAPI specs, so for example

The content on
https://shields.io/badges/npm-downloads
is generated from
https://github.com/badges/shields/blob/cbedd78e42ad4c6f5ab24482771195e6ac64b146/services/npm/npm-downloads.service.js#L54-L74

The content on
https://shields.io/badges/bugzilla-bug-status
is generated from
https://github.com/badges/shields/blob/cbedd78e42ad4c6f5ab24482771195e6ac64b146/services/bugzilla/bugzilla.service.js#L21-L49

...and so on for each of the ~500 services.

**Comment by iach526526** [comment]
Oh, I understand. Synchronizing translations in future versions might be challenging, and the documentation structure could become more complex. Perhaps keeping the documentation as it is now is the better choice. Thank you for your reply.
